### PR TITLE
refactor(Styles): Remove reliance on `css-framework` package

### LINKED ISSUE #434: Remove reliance on deprecated `@defencedigital/css-framework` package
As part of the [related spike](https://github.com/defencedigital/mod-uk-design-system/issues/2894) attempt to remove all reliance on css-framework package within this NextJS application.

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #434 

## Overview

Upgrade dependencies and remove reliance on deprecated `@defencedigital/css-framework` package.

## Reason

Demonstrate that the `css-framework` package is no longer required to consume the other packages within an application, making the codebase v3 ready.

## Work carried out

- [x] Upgrade dependencies to new scope
- [x] Remove reliance on deprecated css-framework
- [x] Add some missing keys
- [x] Fix some 'rules of hooks' lint errors

### CODE REVIEW

File: pages/[...slug].tsx [hunk 1]
```diff
@@ -255,7 +255,7 @@ function renderSidebarItems(
  * Generate `OnThisPage` items based on sub-headings within document
  *
  */
-function renderOnThisPageItems(
+function useRenderOnThisPageItems(
```
Comment (jpveooys) [inline pages/[...slug].tsx:258]: Hmm it's not really being used as a hook though – can it be turned into a component?

File: pages/_app.tsx [hunk 2]
```diff
@@ -1,24 +1,35 @@
...
+import { GlobalStyleProvider } from '@defencedigital/react-component-library'
 import { createGlobalStyle } from 'styled-components'
 
-import '../global.scss'
+const { color } = selectors
 
-const GlobalStyle = createGlobalStyle`
+const AppGlobalStyle = createGlobalStyle`
   html {
     scroll-behavior: smooth;
```
Comment (jpveooys) [inline pages/_app.tsx:12]: In ` global.scss` the larger breakpoint font sizes were being overridden to 100%, so with that gone text becaomes larger as the default breakpoints font sizes are used.

Should we add a `font-size: 100% !important;` here (under `html`) for a similar effect?

**Review by jpveooys — COMMENTED** [review]
Nice work! Couple of small comments.

File: pages/_app.tsx (hunk 2 above)
Comment (m7kvqbe1) [reply pages/_app.tsx:12]: Nice spot I didn't notice that

File: pages/[...slug].tsx (hunk 1 above)
Comment (m7kvqbe1) [reply pages/[...slug].tsx]: Turned it into a functional component.

**Pull request merged**